quinn/src/connection.rs

```diff
@@ -57,7 +57,9 @@ impl Connecting {
             runtime.clone(),
         );
 
-        runtime.spawn(Box::pin(ConnectionDriver(conn.clone())));
+        runtime.spawn(Box::pin(
+            ConnectionDriver(conn.clone()).instrument(Span::current()),
```

Comment:
You mean you avoid long-lived spans? In what way do they cause problems?  If so this would also apply to #1616.

I have used long-living spans freely and not yet seen any issues with them.  Though also the code hasn't been deployed widely or long so I may still be caught out by this.

Reply:
Empirically, this is extremely helpful in our end-to-end tests, and was motivated by difficulty getting clear logs from s user writing similar test cases. In real-world code I think this will provide opportunity for valuable context for quinn events, allowing users to identify *which* connection, in an actionable application-layer sense, is e.g. triggering warnings about malicious behavior.

What are the drawbacks to long lived spans?

Reply:
I think mostly that in some cloud tooling (like OpenTelemetry sending to Google's Cloud Trace), spans only get sent to the visualization backend once it's closed, so a long-lived span will accumulate memory for events in memory until the span ends.

Reply:
Hmm, that seems like a pretty idiosyncratic requirement. For routine "attach context to logging" purposes, long-lived spans are very useful. Are the upstream tracing subscribers receptive to the idea that unbounded memory use for long lived spans is a bug?

Reply:
A user could also arrange to filter spans in their subscriber, though since this will pick up the user's dynamically enclosing spans it might still be a footgun.